### ISSUE TITLE
LDAP login config for grafana?

### ISSUE THREAD

**Issue by KlavsKlavsen** [brancz/kubernetes-grafana#36]
How would I integrate http://docs.grafana.org/installation/ldap/ - config in this ?
It needs to create an extra file - and adjust main config..

**Comment by KlavsKlavsen** [comment]
It seems the grafana/grafana docker image you use - includes ldap.toml:
https://github.com/grafana/grafana-docker/blob/master/Dockerfile

and should be customizable.. I'm digging into how: http://docs.grafana.org/installation/docker/

**Comment by KlavsKlavsen** [comment]
Am I reading this code correct ( https://github.com/brancz/kubernetes-grafana/blob/8f131a315dfe877819c3a490eedfbcfe183b95cf/grafana/grafana.libsonnet#L34 ) - that you build entire grafana.ini based on whats in $._config.grafana.config - and NOT adding anything extra? (ie. no template with default settings)

**Comment by brancz** [comment]
There is no direct support for ldap today in this project, however, in jsonnet you can always dynamically merge the additional file onto the existing objects. I do think a direct support for ldap makes sense though, as it's also a first class citizen in Grafana.

Sadly the ldap configuration is in toml, which is currently not supported in jsonnet. What we could do is have an "ldap" config (I'm thinking `$._config.grafana.ldap`), which contains the full string of the toml configuration. We would default to `null`, and if it's not `null` we create the secret and mount it in the deployment. Would you like to take a stab at this?

**Comment by KlavsKlavsen** [comment]
I would love to do this.. (I need it  - I'm away for 3 weeks though - but when I'm back.. :)
Am I right in my conclusion above - that every config line I need to in grafana.ini - must be in my  $._config.grafana.config ? and if $_config.grafana.config is null - you just don't put it in - and let the docker image use whatever it is born with?

**Comment by KlavsKlavsen** [comment]
btw. I have this in my kube-prometheus config - but I cannot find the word "anonymous" anywhere in my manifests.. Sounds like a bug?
  _config+:: {
    grafana+:: {
      config: {
        sections: {
          "auth.anonymous": {enabled: true},
        },
      },
    },
  },
```

**Comment by brancz** [comment]
> Am I right in my conclusion above - that every config line I need to in grafana.ini - must be in my $._config.grafana.config ? and if $_config.grafana.config is null - you just don't put it in - and let the docker image use whatever it is born with?

Correct.

> I cannot find the word "anonymous" anywhere in my manifests

It's probably because the config is a Secret and encoded in base64.

**Comment by rafregs** [comment]
I've solved this issue through passing multiple files in ``$._config.grafana.config`` variable

```
_config+:: {
  grafana+: {
    config: {
      'grafana.ini': (importstr 'grafana.ini'),
      'ldap.toml': (importstr 'ldap.toml'),
    },
  },
},
```
Both grafana.ini and ldap.toml are both plain files with appropriate configuration.

**Comment by brancz** [comment]
@rafregs previously the grafana.ini was a structured object and only rendered to ini, I think I would like to keep it that way, as we can then merge configurations in downstreams.

**Comment by brancz** [comment]
This was closed by #40
